podpac/core/coordinates/rotated_coordinates.py

```diff
@@ -193,15 +201,19 @@ def __getitem__(self, index):
             origin = self.affine * [I[0], J[0]]
             step = self.step * [index[0].step or 1, index[1].step or 1]
             shape = I.size, J.size
-            return RotatedCoordinates(shape, self.theta, origin, step, **self.properties)
+            return RotatedCoordinates(shape, self.theta, origin, step, dims=self.dims)
 
         else:
-            return super(RotatedCoordinates, self).__getitem__(index)
+            return StackedCoordinates(self.coordinates, dims=self.dims).__getitem__(index)
```

Comment:
This was tricky.

The `StackedCoordinates` is fundamentally a wrapper around its `_coords` attribute, a list of `Coordinates1d`. But the `RotatedCoordinates` doesn't have one because it is parameterized instead. So this really actually is converting from `RotatedCoordinates` to `StackedCoordinates` and then calling `__getitem__` in order to get access to the `_coords` list.

I *could* split `StackedCoordinates` up into `StackedCoordinatesBase`, `StackedCoordinates`, `RotatedCoordinates` (sort of analogous to `Coordinates1d`, `ArrayCoordinates1d', and `UniformCoordinates1d`). That would factor out common code that doesn't rely on the `_coords` attribute and use super when appropriate. Then it would be very clear that this code is meant to convert to `StackedCoordinates`. That would probably be the correct way to do this.

Another alternative that works is to implement `RotatedCoordinates._coords` as a property, and then any methods or super calls that need the `_coords` list would work. But it defeats the purpose of parameterizing the coordinates if you make the full coordinates all the time inadvertantly!

Reply:
I get it. This is fine. The old `super` just tripped me up. Please just comment the code to explain that RotatedCoordinates doesn't have coords and you're essentially creating them here, and then accessing through stacked coordinate.s